### ISSUE TITLE
500 error on new NextJS app after successful build and deploy

### ISSUE THREAD

**Issue by bflanagan2** [aws-amplify/amplify-hosting#3479]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d1g7ti7hmwjxjd

### AWS Region

us-east-1

### Amplify Hosting feature

Deployments, Frontend builds

### Describe the bug

I've added a new NextJS app to Amplify using my normal procedure in the UI using defaults as I've done for several other apps but for this one, after a successful build and deploy, the generated URL returns a 500 error.  There seems to be two log entries that might be relevant: one is in the Clone step and the other in the Frontend step. 

### Expected behavior

Site renders as expected.

### Reproduction steps

1. Select "Host Web App" from the "New app" drop down.
2. Connect github repo
3. Assign branch (master)
4. Accept build settings and click "Next"
5. Click "Save and Deploy"
6. Wait for build and deploy to complete.
7. Click on hyperlink to assigned URL
8. Received 500 error

### Build Settings

```yaml
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

### Log output

<details>

```
# Put your logs below this line
#Clone 
2023-05-11T18:58:51.022Z [INFO]: # Retrieving environment cache...
2023-05-11T18:58:51.075Z [WARNING]: ! Unable to write cache: {"code":"ERR_BAD_REQUEST","message":"Request failed with status code 404"})}
2023-05-11T18:58:51.075Z [INFO]: ---- Setting Up SSM Secrets ----
2023-05-11T18:58:51.076Z [INFO]: SSM params {"Path":"/amplify/d1g7ti7hmwjxjd/master/","WithDecryption":true}
2023-05-11T18:58:51.110Z [WARNING]: !Failed to set up process.env.secrets

#Frontend 
2023-05-11T18:59:45.886Z [INFO]: info @next/swc-darwin-arm64@13.4.1: The platform "linux" is incompatible with this module.
2023-05-11T18:59:45.889Z [INFO]: info "@next/swc-darwin-arm64@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-darwin-arm64@13.4.1: The CPU architecture "x64" is incompatible with this module.
                                 info @next/swc-darwin-x64@13.4.1: The platform "linux" is incompatible with this module.
                                 info "@next/swc-darwin-x64@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-linux-arm64-gnu@13.4.1: The CPU architecture "x64" is incompatible with this module.
                                 info "@next/swc-linux-arm64-gnu@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-linux-arm64-musl@13.4.1: The CPU architecture "x64" is incompatible with this module.
                                 info "@next/swc-linux-arm64-musl@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-win32-arm64-msvc@13.4.1: The platform "linux" is incompatible with this module.
                                 info "@next/swc-win32-arm64-msvc@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-win32-arm64-msvc@13.4.1: The CPU architecture "x64" is incompatible with this module.
                                 info @next/swc-win32-ia32-msvc@13.4.1: The platform "linux" is incompatible with this module.
                                 info "@next/swc-win32-ia32-msvc@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info @next/swc-win32-ia32-msvc@13.4.1: The CPU architecture "x64" is incompatible with this module.
                                 info @next/swc-win32-x64-msvc@13.4.1: The platform "linux" is incompatible with this module.
                                 info "@next/swc-win32-x64-msvc@13.4.1" is an optional dependency and failed compatibility check. Excluding it from installation.
                                 info fsevents@2.3.2: The platform "linux" is incompatible with this module.
                                 info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
```

</details>

**Comment by bflanagan2** [comment]
**Update** Looks like this was introduced with a recent NextJS minor version. I downgraded to next@13.1.6 and the site builds as expected. (Not yet sure which version introduced the issue yet though)

**Comment by terodox** [comment]
NextJS had a breaking change in 13.4.0 and 13.4.1 that was fixed in 13.4.2

Upgrading to 13.4.2 or higher should resolve this issue.

Original issue link: https://github.com/aws-amplify/amplify-hosting/issues/3466
